### ISSUE TITLE
Padding in CTC Loss

### ISSUE THREAD

**Issue by jessling4711-lang** [keras-team/keras#21876]
Hey, 

while migrating an CRNN of a production OCR from tentoflow/keras to keras 3, we just found an issue with the CTC-loss implementation. There is not way, to proper ignore padding in the new backed which leads to pretty bad results. The old version (backend.ctc_batch_cost) has used -1. The example [https://keras.io/examples/audio/ctc_asr/](url) suggest, that zero is used as padding (the blank label index) in the new version. This should not be done as described in [https://stackoverflow.com/questions/49063938/padding-labels-for-tensorflow-ctc-loss](ur). Using a batch-size of 1, fixes the broken example [https://keras.io/examples/audio/ctc_asr/](url).

We are using tensorflow 2.20 as backend and keras 3.12.0

Thank you.
